django-verdant/rca/management/commands/archive_events.py

```diff
@@ -0,0 +1,161 @@
+from django.db import IntegrityError, transaction
+from rca.models import EventItem, EventIndex
+from wagtail.wagtailredirects.models import Redirect
+
+# Ignore all the deprecation warnings.
+# Legacy site, soon to be replaced.
+warnings.simplefilter("ignore")
+
+
+class Command(BaseCommand):
+    event_redirects = {}
+
+    option_list = BaseCommand.option_list + (
+        make_option(
+            '--dry-run',
+            action='store_true',
+            dest='dry_run',
+            default=False,
+            help="Dry run, do not unpublish pages or create redirects."
+        ),
+        make_option(
+            '--verbose',
+            action='store_true',
+            dest='verbose',
+            default=True,
+            help="Provide extended feedback."
+        ),
+        make_option(
+            '--examples',
+            action='store_true',
+            dest='examples',
+            default=False,
+            help="Provide urls for example event content."
+        ),
+        make_option(
+            '--csv',
+            action='store_true',
+            dest='output_csv',
+            default=False,
+            help="List archived event items. Negates verbose."
+        ),
+        make_option(
+            '--permanent',
+            action='store_true',
+            dest='permanent',
+            default=True,
+            help="Create permanent redirects (301)."
+        ),
+    )
+
+    def create_redirect(self, e, target_page, permanent=True):
+        new_redirect = Redirect()
+        new_redirect.site = e.get_site()
+        # Path includes 'home' and a trailing slash
+        # both must be removed.
+        bare_path = '/' + '/'.join(e.url_path.split('/')[2:][:-1])
+        new_redirect.old_path = bare_path
+        new_redirect.is_permanent = permanent
+        new_redirect.redirect_page = target_page
+        new_redirect.save()
+        self.event_redirects[e.pk] = new_redirect.id
+
+    def get_csv_rows(self, events):
+        rows = []
+        headers = [
+            'Event id',
+            'Event title',
+            'Event URL',
+            'Dates from',
+            'Dates to',
+            'Redirect',
+        ]
+        rows.append(headers)
+        for e in events:
+            row = []
+            row.append(e.pk)
+            row.append(e.title.encode('utf8'))
+            row.append(e.url)
+            dates = e.dates_times.order_by('date_from').values_list('date_from', 'date_to')
+            date_from = []
+            for d in dates:
+                try:
+                    date_from.append(d[0].strftime("%d/%m/%Y"))
+                except Exception:
+                    date_from = ''
+
+            date_to = []
+            for d in dates:
+                try:
+                    date_to.append(d[1].strftime("%d/%m/%Y"))
+                except Exception:
+                    date_to = ''
+
+            row.append(', '.join(date_from))
```

Comment:
Is this adding a comma separated list of dates to a CSV? If so I'd consider using a `|` or something else that could cause issues in opening the CSV.

Reply:
Oops, yep, thats a really good call